sormas-rest/src/main/java/de/symeda/sormas/rest/security/oidc/S2SAuthFilter.java

```diff
@@ -0,0 +1,128 @@
+package de.symeda.sormas.rest.security.oidc;
+
+import java.io.IOException;
+import java.net.URL;
+import java.security.PublicKey;
+
+import javax.annotation.Priority;
+import javax.ws.rs.HttpMethod;
+import javax.ws.rs.Priorities;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.Provider;
+
+import org.glassfish.jersey.server.ContainerRequest;
+import org.keycloak.TokenVerifier;
+import org.keycloak.common.VerificationException;
+import org.keycloak.jose.jwk.JSONWebKeySet;
+import org.keycloak.jose.jwk.JWK;
+import org.keycloak.jose.jwk.JWKParser;
+import org.keycloak.representations.AccessToken;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import de.symeda.sormas.api.FacadeProvider;
+import de.symeda.sormas.api.SormasToSormasConfig;
+import de.symeda.sormas.api.sormastosormas.SormasToSormasEncryptedDataDto;
+
+@Provider
+@Priority(Priorities.AUTHENTICATION)
+@ClientCredentials
+public class S2SAuthFilter implements ContainerRequestFilter {
+
+	private final Logger LOGGER = LoggerFactory.getLogger(getClass());
+	private static final String BEARER = "Bearer";
+
+	@Override
+	public void filter(ContainerRequestContext requestContext) {
+		String authorizationHeader = requestContext.getHeaderString(HttpHeaders.AUTHORIZATION);
+		boolean validHeader = authorizationHeader != null && authorizationHeader.startsWith(String.format("%s ", BEARER));
+
+		if (!validHeader) {
+			requestContext.abortWith(Response.status(Response.Status.UNAUTHORIZED.getStatusCode(), "Invalid header").build());
```

Comment:
Should maybe also stop the execution of the method here, as it might result in NPE

Reply:
absolutely :)